pkg/generate/code/set_resource.go

```diff
@@ -1257,26 +1328,71 @@ func setResourceForSlice(
 		targetShape.MemberRef.Shape,
 		indentLevel+1,
 	)
-	//  f0elem0 = *f0iter0
+	// We may have some instructions to specially handle this field by
+	// accessing a different Output shape member.
 	//
-	// or
+	// As an example, let's say we're dealing with a resource called
+	// DBInstance that has a "DBSecurityGroups" field of type `[]*string`.
+	// In the Output shape of the ReadOne resource manager method, the
+	// field name is "DBSecurityGroups" of type
+	// `[]*DBSecurityGroupMembership`, though.  This
+	// `DBSecurityGroupMembership` struct contains two fields,
+	// `DBSecurityGroupName` and `DBSecurityGroupStatus` and we wish to
+	// grab only the value of the `DBSecurityGroupName` fields and set the
+	// `Spec.DBSecurityGroups` field to the set of those field values.
 	//
-	//  f0elem0.SetMyField(*f0iter0)
-	containerFieldName := ""
-	if sourceShape.MemberRef.Shape.Type == "structure" {
-		containerFieldName = targetFieldName
+	// There may be a generator.yaml file that looks like this:
+	//
+	// resources:
+	//   DBInstance:
+	//     fields:
+	//       DBSecurityGroups:
+	//         set:
+	//          - method: ReadOne
+	//            from: DBSecurityGroupName
```

Comment:
nit: Not a huge fan of `DBSecurityGroups` being used to reference both CRD field and Output Shape field, but can see why you can't do a syntactically correct path for `from` value. Since you are resolving a type mismatch issue here, what if the setconfig `from` reflected the path to the value of a given _type_  -- not the exact field name in Output Shape? Something like:

```
	// resources:
	//   DBInstance:
	//     fields:
	//       DBSecurityGroups:
	//         set:
	//          - method: ReadOne
	//            from_type: DBSecurityGroupMembership.DBSecurityGroupName
```